std/bigint.d

```diff
@@ -1082,16 +1082,16 @@ Returns:
     A $(D string) that represents the $(D BigInt) as a decimal number.
 
 */
-string toDecimalString(const(BigInt) x)
+string toDecimalString(const(BigInt) x) pure
 {
-    string outbuff="";
-    void sink(const(char)[] s) { outbuff ~= s; }
-    x.toString(&sink, "%d");
-    return outbuff;
+    auto buff = x.data.toDecimalString(0);
+
+    if (x.isNegative()) return '-' ~ buff;
+    return buff;
```

Comment:
`BigUint.toDecimalString` takes a parameter telling it how much "extra" padding to put at the front. I think this is specifically to allow adding a '-' sign. We should take advantage of that here:

```D
auto buff = x.data.toDecimalString(x.isNegative() ? 1 : 0);
if(x.isNegative())
    buff[0] = '-';
```